rotkehlchen/chain/evm/decoding/gearbox/balances.py

```diff
@@ -0,0 +1,89 @@
+import logging
+from collections import defaultdict
+from typing import TYPE_CHECKING
+
+from rotkehlchen.accounting.structures.balance import Balance, BalanceSheet
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.chain.ethereum.interfaces.balances import BalancesSheetType, ProtocolWithBalance
+from rotkehlchen.chain.ethereum.utils import token_normalized_value_decimals
+from rotkehlchen.chain.evm.constants import DEFAULT_TOKEN_DECIMALS
+from rotkehlchen.chain.evm.contracts import EvmContract
+from rotkehlchen.chain.evm.decoding.gearbox.constants import CPT_GEARBOX
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.history.events.structures.evm_event import EvmProduct
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.inquirer import Inquirer
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.db.dbhandler import DBHandler
+    from rotkehlchen.types import ChecksumEvmAddress
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class GearboxCommonBalances(ProtocolWithBalance):
+    def __init__(
+            self,
+            database: 'DBHandler',
+            evm_inquirer: 'EvmNodeInquirer',
+            staking_contract: 'ChecksumEvmAddress',
+            native_token_id: str,
+    ):
+        super().__init__(
+            database=database,
+            evm_inquirer=evm_inquirer,
+            counterparty=CPT_GEARBOX,
+            deposit_event_types={(HistoryEventType.STAKING, HistoryEventSubType.DEPOSIT_ASSET)},
+        )
+        self.native_token_id = native_token_id
+        self.staking_contract = staking_contract
+
+    def query_balances(self) -> 'BalancesSheetType':
+        """Query balances of staked gear tokens if deposit events are found."""
+        balances: BalancesSheetType = defaultdict(BalanceSheet)
+
+        # fetch deposit events
+        if len(addresses_with_deposits := list(self.addresses_with_deposits(products=[EvmProduct.STAKING]))) == 0:  # noqa: E501
+            return balances
+
+        calls_mapping = list(addresses_with_deposits)
+        calls = [
+            (
+                (_contract := EvmContract(
+                    address=self.staking_contract,
+                    abi=self.evm_inquirer.contracts.abi('GEARBOX_STAKING'),
```

Comment:
this is a trip to the db for each user. Why do it? Instead initialize the contract once outside the loop and re-use it